trin-core/src/portalnet/events.rs

```diff
@@ -33,68 +49,113 @@ impl PortalnetEvents {
         Self {
             discovery: Arc::clone(&discovery),
             protocol_receiver,
-            history_sender,
-            state_sender,
-            utp_sender,
+            utp_listener_receiver,
+            history_overlay_sender,
+            history_utp_sender,
+            state_overlay_sender,
+            state_utp_sender,
+            utp_listener_sender,
         }
     }
 
-    /// Receives a request from the talkreq handler and sends a response back
-    pub async fn process_discv5_requests(mut self) {
-        while let Some(event) = self.protocol_receiver.recv().await {
-            let request = match event {
-                Discv5Event::TalkRequest(r) => r,
-                Discv5Event::NodeInserted { node_id, replaced } => {
-                    match replaced {
-                        Some(old_node_id) => {
-                            debug!(
-                                "Received NodeInserted(node_id={}, replaced={})",
-                                node_id, old_node_id,
-                            );
-                        }
-                        None => {
-                            debug!("Received NodeInserted(node_id={})", node_id);
+    /// Main loop to dispatch `Discv5` and `UtpListener` events to overlay networks
+    pub async fn start(mut self) {
+        loop {
+            tokio::select! {
```

Comment:
This select is huge. I expect it to be much easier to read if it calls some functions here at each of the branches. Even some of the branches seem large enough to break down into a couple functions.

I don't feel confident about having reviewed this section until that happens.

Reply:
Refactored this a bit, it should be more readable now.